packages/components/radio-group/src/RadioGroup.stories.tsx

```diff
@@ -116,6 +119,118 @@ export const Disabled: StoryFn = _args => (
   </RadioGroup>
 )
 
+export const CustomImplementation: StoryFn = () => {
+  const CustomRadio = ({
+    children,
+    selectedValue,
+    ...others
+  }: RadioProps & { selectedValue: string }) => {
+    const id = useId()
+    const { value } = others
+
+    return (
+      <Label
+        id={id}
+        htmlFor={value}
+        className={cx(
+          'flex flex-wrap shadow rounded-md gap-md p-lg max-w-sz-320',
+          value === selectedValue ? 'bg-success/dim-4' : '',
+          'cursor-pointer'
+        )}
+      >
+        <RadioGroup.Radio aria-labelledby={id} id={value} {...others} />
+        {children}
+      </Label>
+    )
+  }
+
+  const Example = () => {
+    const [value, setValue] = useState<string>('')
+
+    function onValueChange(current: string) {
+      setValue(current)
+    }
+
+    const radios = ['A', 'B', 'C']
+
+    return (
+      <RadioGroup value={value} name="sport" onValueChange={onValueChange}>
+        {radios.map(radio => {
+          return (
+            <CustomRadio selectedValue={value} key={radio} value={radio}>
+              <div className="flex grow justify-between">
+                <span className="font-bold">{radio}</span>
+                <span>this is a custom</span>
+              </div>
+              <div className="w-full text-right italic">implementation of a radio</div>
+            </CustomRadio>
+          )
+        })}
+      </RadioGroup>
+    )
+  }
+
+  return <Example />
+}
+
+export const InvisibleRadioGroup: StoryFn = () => {
+  const CustomRadio = ({
+    children,
+    selectedValue,
+    ...others
+  }: RadioProps & { selectedValue: string }) => {
+    const id = useId()
+    const { value } = others
+
+    return (
+      <Label
+        id={id}
+        htmlFor={value}
+        className={cx(
+          'flex flex-wrap shadow rounded-md gap-md p-lg max-w-sz-320',
+          value === selectedValue ? 'bg-success/dim-4' : '',
```

Comment:
Not sure if we should write some agreement about this kind of logic when defining classnames 🤔

Reply:
What kind of agreements do you have in mind?